R/csv.R

```diff
@@ -149,76 +149,14 @@ pl$scan_csv = function(
-#' @param dtypes Named list of column names - dtypes or dtype - column names. This
-#' list is used while reading to overwrite dtypes. Supported types so far are:
-#' * "Boolean" or "logical" for DataType::Boolean,
-#' * "Categorical" or "factor" for DataType::Categorical,
-#' * "Float32" or "double" for DataType::Float32,
-#' * "Float64" or "float64" for DataType::Float64,
-#' * "Int32" or "integer" for DataType::Int32,
-#' * "Int64" or "integer64" for DataType::Int64,
-#' * "Utf8" or "character" for DataType::Utf8,
-#' @param null_values Values to interpret as `NA` values. Can be:
-#' * a character vector: all values that match one of the values in this vector
-#'   will be `NA`;
-#' * a named list with column names and null values.
-# @param missing_utf8_is_empty_string By default, a missing value is considered
-# to be `NA`. Setting this parameter to `TRUE` will consider missing UTF8 values
-# as an empty character.
-#' @param ignore_errors Keep reading the file even if some lines yield errors.
-#' You can also use `infer_schema_length = 0` to read all columns as UTF8 to
-#' check which values might cause an issue.
-#' @param cache Cache the result after reading.
-#' @param infer_schema_length Maximum number of rows to read to infer the column
-#' types. If set to 0, all columns will be read as UTF-8. If `NULL`, a full
-#' table scan will be done (slow).
-#' @param n_rows Maximum number of rows to read.
-#' @param encoding Either `"utf8"` or `"utf8-lossy"`. Lossy means that invalid
-#' UTF8 values are replaced with "?" characters.
-#' @param low_memory Reduce memory usage (will yield a lower performance).
-#' @param rechunk Reallocate to contiguous memory when all chunks / files are
-#' parsed.
-#' @param skip_rows_after_header Parse the first row as headers, and then skip
-#' this number of rows.
-#' @param row_count_name If not `NULL`, this will insert a row count column with
-#' the given name into the DataFrame.
-#' @param row_count_offset Offset to start the row_count column (only used if
-#' the name is set).
-#' @param try_parse_dates Try to automatically parse dates. Most ISO8601-like
-#' formats can be inferred, as well as a handful of others. If this does not
-#' succeed, the column remains of data type `pl$Utf8`.
-#' @param eol_char Single byte end of line character (default: `\n`). When
-#' encountering a file with Windows line endings (`\r\n`), one can go with the
-#' default `\n`. The extra `\r` will be removed when processed.
-#' @param raise_if_empty If `FALSE`, parsing an empty file returns an empty
-#' DataFrame or LazyFrame.
-#' @param truncate_ragged_lines Truncate lines that are longer than the schema.
-#' @param reuse_downloaded If `TRUE`(default) and a URL was provided, cache the
-#' downloaded files in session for an easy reuse.
+#' @inheritParams scan_csv
 #' @return DataFrame
-pl$read_csv = function(
+read_csv = function( # remapped to pl$read_csv, a hack to support roxygen2 @inheritsParams
```

Comment:
cool hack, nice to get rid of all these duplicated docs

Reply:
I thinks this is related to #560 and the function name  should be `pl_read_csv`.
This function should be add to the `pl` environment when loading.